signals.go

```diff
@@ -30,7 +35,9 @@ func newSignalHandler(enableSubreaper bool) *signalHandler {
 	// handle all signals for the process.
 	signal.Notify(s)
 	return &signalHandler{
-		signals: s,
+		signals:          s,
+		notifySocket:     notifySocket,
```

Comment:
can you create a struct for this, we have the two fields and are passing them around everywhere.  A struct will clean up the code alot and you can add methods for the socket there.